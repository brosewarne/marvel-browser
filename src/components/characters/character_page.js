import React from 'react';
import PropTypes from 'prop-types';

import { CharacterImage } from './character_image';

import './character.css';

/**
 * Render the CharacterPage, showing a profile pof the selected character
 * @param {Object} props - The set of props for the CharacterPage - passed from the route
 * @returns {Node} - The rendered CharacterPage
 * @constructor
 */
export const CharacterPage = (props) => {
    const { character } = props.location.state;
    const {
        name, description, comics, series, stories
    } = character;

    return (
        <div>
            <content className="characterPage">
                <div className="characterProfile">
                    <h1 className="characterProfileTitle">{name}</h1>
                    <CharacterImage character={character} imgSize="portrait_uncanny" />
                    <p>{description || 'No character description available'}</p>
                    <p>{`${comics.available} Comics`}</p>
                    <p>{`${series.available} Series`}</p>
                    <p>{`${stories.available} Stories`}</p>
                </div>
            </content>
        </div>
    );
};

CharacterPage.propTypes = {
    location: PropTypes.object.isRequired
};
