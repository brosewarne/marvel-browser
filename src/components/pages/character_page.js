import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Paper from 'material-ui/Paper';
import Typography from 'material-ui/Typography';

import { getDetailLinks } from './utils';
import { ItemImage } from '../items/item_image';

import './pages.css';

/**
 * Render the CharacterPage, showing a profile of the selected character
 * @param {Object} props - The set of props for the CharacterPage - passed from the route
 * @returns {Node} - The rendered CharacterPage
 * @constructor
 */
export const _CharacterPage = (props) => {
    const {
        name, description, comics, series, stories, urls
    } = props.item;

    return (
        <div>
            <content className="pageContainer">
                <Paper elevation={12} className="itemProfile">
                    <h1 className="itemProfileTitle">{name}</h1>
                    <ItemImage item={props.item} imgSize="portrait_uncanny" />
                    <Typography align="center" variant="body2">
                        <p>{description || 'No character description available'}</p>
                        <p>{`${comics.available} Comics`}</p>
                        <p>{`${series.available} Series`}</p>
                        <p>{`${stories.available} Stories`}</p>
                    </Typography>
                    {getDetailLinks(props.item)}
                </Paper>
            </content>
        </div>
    );
};

_CharacterPage.propTypes = {
    item: PropTypes.object.isRequired
};

/**
 * If item is not supplied from props, get it from the app state
 * @param {Object} state - The current app state
 * @param {Object} props - The incoming set of props
 * @returns {Object} - Empty id item is already in props, otherwise, the mapped item from app state
 */
const mapStateToProps = (state, props) => {
    return props.item ? {} : {
        item: state.routing.location.state.item
    };
};

export const CharacterPage = connect(mapStateToProps)(_CharacterPage);
