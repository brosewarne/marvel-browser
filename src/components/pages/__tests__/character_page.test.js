import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { _CharacterPage } from '../character_page';

configure({ adapter: new Adapter() });

describe('<_CharacterPage />', () => {
    const mockCharacter = {
        id: 1009146,
        name: 'Abomination',
        thumbnail: {
            path: '/a/href/to/an/image',
            extension: 'png'
        },
        description: 'He\'s an abomination',
        comics: {
            available: 10
        },
        series: {
            available: 2
        },
        stories: {
            available: 5
        },
        urls: [
            {
                url: '/aURL',
                type: 'wiki'
            }
        ]
    };
    it('renders correctly', () => {
        const wrapper = shallow(
            <_CharacterPage item={mockCharacter} />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
