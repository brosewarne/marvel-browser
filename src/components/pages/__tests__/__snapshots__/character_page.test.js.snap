// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<_CharacterPage /> renders correctly 1`] = `
<div>
  <content
    className="pageContainer"
  >
    <WithStyles(Paper)
      className="itemProfile"
      elevation={12}
    >
      <h1
        className="itemProfileTitle"
      >
        Abomination
      </h1>
      <Component
        imgSize="portrait_uncanny"
        item={
          Object {
            "comics": Object {
              "available": 10,
            },
            "description": "He's an abomination",
            "id": 1009146,
            "name": "Abomination",
            "series": Object {
              "available": 2,
            },
            "stories": Object {
              "available": 5,
            },
            "thumbnail": Object {
              "extension": "png",
              "path": "/a/href/to/an/image",
            },
            "urls": Array [
              Object {
                "type": "wiki",
                "url": "/aURL",
              },
            ],
          }
        }
      />
      <WithStyles(Typography)
        align="center"
        variant="body2"
      >
        <p>
          He's an abomination
        </p>
        <p>
          10 Comics
        </p>
        <p>
          2 Series
        </p>
        <p>
          5 Stories
        </p>
      </WithStyles(Typography)>
      <WithStyles(Button)
        color="secondary"
        variant="raised"
      >
        <Link
          href="/aURL"
          replace={false}
          target="_blank"
          to="/aURL"
        >
          Wiki
        </Link>
      </WithStyles(Button)>
    </WithStyles(Paper)>
  </content>
</div>
`;
